### DISCUSSION TITLE
[Feature Request] NextUI integration in React Native

### DISCUSSION THREAD

**Discussion by fabrizioanichini** [heroui-inc/heroui#592] [Feature requests]
### Is your feature request related to a problem? Please describe.

Is there any way to do that now?

### Describe the solution you'd like

As a user I would like to use the UI system of NextUI in React Native apps.

### Describe alternatives you've considered

The other UI frameworks in React Natvie are ugly

**Comment by dohomi** [comment]
I'd love to see this implemented.

**Comment by ansmlc** [comment]
Please, yes. There is nothing comparable to NextUI in React Native.

**Comment by 0xAskar** [comment]
Any updates on this?

**Comment by olehmelnyk** [comment]
hey, have you heard about mitosis? https://github.com/BuilderIO/mitosis
It looks like it can convert JSX-like components to other frameworks, including React Native, Angular, Vue, etc. - so MAYBE with it - it would be possible to make NextUI framework agnostic

**Comment by cvolant** [comment]
To me, there are 2 main issues:
- Tailwind
- Framer Motion

Nativewind can be use in react-native instead of Tailwind. But there is still Framer Motion. If ever Moti could replace Framer Motion, then using nextui in React native would be nearly possible.

**Reply by akhockey21** [reply]
I use NativeWind and it works perfect with any tailwind configurations you want.

**Reply by therealsras** [reply]
> I use NativeWind and it works perfect with any tailwind configurations you want.

Do you intergate with nextui and RN

**Comment by 0xAskar** [comment]
@cvolant yeah, I was going to say that both nativewind and moti could replace them right? does moti not cover framer in whole?

**Comment by cvolant** [comment]
Probably not entirely, but maybe most of.
But I forgot something : react-aria.
As I was looking for an alternative to react-aria, I discovered [Gluestack](https://gluestsck.io): interesting lib that could do the whole job, it seems not as complete and beautiful as NextUI, but it is built for React Native, Expo, Next and server components.

**Comment by alcalawil** [comment]
I this in this still under discussion? I would love to have these components in my react native app

**Comment by MoySantanaV** [comment]
I hope we can get NextUI in React Native soon!

**Comment by covelitein** [comment]
NextUI is a very powerful library, I will be really happy if the library incorporates it's Toast notifications

**Reply by JavierTF** [reply]
![Captura de pantalla 2025-01-04 190826](https://github.com/user-attachments/assets/0f5ea6f5-5eb0-40d8-a63f-29b370480f45)
Coming soon

**Comment by RubenStark** [comment]
That would be awesome!

**Comment by prince272** [comment]
It seems this is never going to happen as it was not intended before.

**Reply by therealsras** [reply]
> It seems this is never going to happen as it was not intended before.

it will come soon 
![image](https://github.com/user-attachments/assets/cf35507d-9d73-4b55-9950-d205055093ab)

**Reply by Mahmoud-walid** [reply]
I will use it immediately with React Native

**Comment by alpppine** [comment]
I'm thrilled to see this is on the HeroUI roadmap!!!! Does anyone have a rough idea when this might be released? cc @jrgarciadev

**Comment by pouyajabbarisani** [comment]
looking forwaed to see this

**Comment by prince272** [comment]
Looking forward to also seeing this too....

**Comment by Moka-admin0** [comment]
Looking forward to also seeing this too.... !!

**Comment by Malberee** [comment]
If anyone finds it useful, I have ported some HeroUI components to React Native. I did this for myself, but you can try using it too. In any case, feel free to create issues if you encounter any problems.
https://github.com/Malberee/heroui-native

**Comment by devOrv876** [comment]
+1
this is needed !

Native wind and Reanimated

**Comment by osarhan** [comment]
Its here! https://github.com/heroui-inc/heroui-native

**Comment by devOrv876** [comment]
> Its here! https://github.com/heroui-inc/heroui-native

Its Beautiful 😍

**Comment by kzroo** [comment]
im impressed
